### PR TITLE
Configurable subjective account decay time

### PR DESCRIPTION
**Pull request by ClaytonCalabrese**
### Adds new config option subjective-account-decay-time-minutes to Producer Plugin

This value is used to calculate the expired_accumulator_average_window, thus making account decay time configurable. The passed value must be above 0, and defaults to 1440.

The default value for expired_accumulator_average_window remains the same.

### CODE REVIEW

**Comment by matthewdarwin** [comment]
Can this be changed to >= 30 minutes?  (I am testing 30 in production right now)

File: plugins/producer_plugin/test/test_subjective_billing.cpp [hunk 1]
```diff
@@ -129,7 +131,7 @@ BOOST_AUTO_TEST_CASE( subjective_bill_test ) {
 
    { // expired handling logic, full billing until expiration then failed/decay logic
       subjective_billing sub_bill;
-      constexpr uint32_t window_size = subjective_billing::expired_accumulator_average_window;
+      uint32_t window_size = sub_bill.get_expired_accumulator_average_window();
```
Comment (heifner) [inline plugins/producer_plugin/test/test_subjective_billing.cpp:134]: This is not used, so just remove it.

**Review by heifner — CHANGES_REQUESTED** [review]

File: plugins/producer_plugin/include/eosio/producer_plugin/subjective_billing.hpp [hunk 2]
```diff
@@ -204,6 +204,15 @@ class subjective_billing {
       }
       return !exhausted;
    }
+
+   uint32_t get_expired_accumulator_average_window() {
```
Comment (heifner) [inline plugins/producer_plugin/include/eosio/producer_plugin/subjective_billing.hpp:208]: Make method `const`

File: plugins/producer_plugin/producer_plugin.cpp [hunk 3]
```diff
@@ -917,12 +917,10 @@ void producer_plugin::plugin_initialize(const boost::program_options::variables_
...
 
-   uint32_t subjective_account_decay_time_minutes = options.at( "subjective-account-decay-time-minutes" ).as<uint32_t>();
-   EOS_ASSERT( subjective_account_decay_time_minutes > 30, plugin_config_exception,
-               "subjective-account-decay-time-minutes ${dt} must be greater than 30", ("dt", subjective_account_decay_time_minutes));
-   if( subjective_account_decay_time_minutes != config::account_cpu_usage_average_window_ms / 1000 / 60 ) {
-      my->_subjective_billing.set_expired_accumulator_average_window( subjective_account_decay_time_minutes );
-   }
+   fc::microseconds subjective_account_decay_time = fc::minutes(options.at( "subjective-account-decay-time-minutes" ).as<uint32_t>());
+   EOS_ASSERT( subjective_account_decay_time.count() > 0, plugin_config_exception,
+               "subjective-account-decay-time-minutes ${dt} must be greater than 0", ("dt", subjective_account_decay_time.to_seconds() * 60));
```
Comment (heifner) [inline plugins/producer_plugin/producer_plugin.cpp:922]: `/ 60` not `* 60`

**Review by heifner — APPROVED** [review]

**Pull request merged**